From 426870780fb076405953a1e746dc6903f66efc53 Mon Sep 17 00:00:00 2001
From: Hecheng Yu <kde-yyds@qq.com>
Date: Sat, 1 Aug 2026 14:13:45 +0800
Subject: [PATCH 6/8] dodge: Clamp 720p capture rate to 30fps to fix 7.5fps
 recording

The 4x-slow-motion problem of the 8K fix applies to 720p as well: after
switching 4K120 -> 720p, CommonVideoMode.li() returns 120.0 (stale fps
state) and MediaRecorder.setCaptureRate(120) is sent while the encoder
---
 .../com/oplus/camera/module/a.smali           | 22 +++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/smali_classes12/com/oplus/camera/module/a.smali b/smali_classes12/com/oplus/camera/module/a.smali
index 4c12e49a..246c0712 100644
--- a/smali_classes12/com/oplus/camera/module/a.smali
+++ b/smali_classes12/com/oplus/camera/module/a.smali
@@ -31932,6 +31932,28 @@
 
     :cond_a12_ri_8k
 
+    invoke-virtual {p0}, Lek/u0;->F6()Ljava/lang/String;
+
+    move-result-object v0
+
+    const-string v1, "video_size_720p"
+
+    invoke-virtual {v1, v0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+
+    move-result v0
+
+    if-eqz v0, :cond_a12_ri_base
+
+    const-wide/high16 v0, 0x403e000000000000L
+
+    invoke-static {v0, v1}, Ljava/lang/Double;->valueOf(D)Ljava/lang/Double;
+
+    move-result-object p0
+
+    return-object p0
+
+    :cond_a12_ri_base
+
     .line 1
     invoke-virtual {p0}, Lcom/oplus/camera/module/a;->r()Z
 
-- 
2.55.0

